incubator/hnc/Makefile

```diff
@@ -22,7 +22,11 @@ PROJECT_ID := ${shell gcloud config get-value project 2>/dev/null}
 # By default, we use the Container Registry in the current GCP project, but you
 # can set it to anything, UNLESS you are calling 'make release' since the image
 # needs to be built in the same project as the GCB instance that builds it.
-HNC_REGISTRY ?= gcr.io/${PROJECT_ID}
+ifdef PROJECT_ID
+  HNC_REGISTRY ?= gcr.io/${PROJECT_ID}
+else
+  HNC_REGISTRY ?= gcr.io
+endif
```

Comment:
`gcr.io` on its own isn't really a registry. How about this:

* If `PROJECT_ID` doesn't exist, `HNC_REGISTRY` is an empty string
* On line 51, if `HNC_REGISTRY` is undefined, we just omit it when creating `HNC_IMG`.

E.g., if `PROJECT_ID` is my-proj, then `HNC_IMG` will be `gcr.io/my-proj/hnc-manager:kind-local` by default (if we're configured for KIND). But if `PROJECT_ID` is missing, then `HNC_IMG` will be `hnc-manager:kind-local` which is a perfectly valid image name for an image that will never leave the current computer.

I know it's a bit more work but it prevents us from having a registry that _looks_ valid but doesn't actually mean anything. Does this make sense?

Reply:
Yes, definitely it does.

Reply:
Make the requested changes.